tests/unit/test_agent_skill.py

```diff
@@ -1631,3 +1633,130 @@ def test_lint_file_fail_typescript(tmp_path, capsys):
                 assert result_line.lstrip('./') == expected_line.lstrip(
                     './'
                 ), f"Line {i+1} doesn't match"
+
+
+def test_edit_file_by_replace_with_linebreaks(tmp_path):
+    temp_file_path = tmp_path / 'linebreak_test.txt'
+    content = 'Line 1\\n\nLine 2\r\nLine 3\\r\\n\nLine 4\n"Docstring\\n with\\r\\n linebreaks"'
+    temp_file_path.write_text(content)
+
+    open_file(str(temp_file_path))
+
+    with io.StringIO() as buf:
+        with contextlib.redirect_stdout(buf):
+            edit_file_by_replace(
+                file_name=str(temp_file_path),
+                to_replace='Line 2\r\nLine 3\\r\\n',
+                new_content='New Line 2\\nNew Line 3\r\n',
+            )
+        result = buf.getvalue()
+        expected = (
+            f'[File: {temp_file_path} (4 lines total after edit)]\n'
+            '(this is the beginning of the file)\n'
+            '1|Line 1\\n\n'
+            '2|Line 2\n'
```

Comment:
Shouldn't this `Line 2` get replaced? because `to_replace='Line 2\r\nLine 3\\r\\n',`

Reply:
Actually not in this case as the search term is `Line 2\r\nLine 3\\r\\n` (note the `\r`)